<!-- Launch the rover base controller node

Parameters
    classifier_filename : str
        The full filepath for the `scikit-learn` classifier model.
    classifier_window : int
        The size of the classifier window, this sets the number of
        entries in the servo history used to train the classifier.
        The classifier and regressor models must correspond to this
        setting. (default 10)
    regressor_filename : str
        The full filepath for the `scikit-learn` regressor model.
    model : str
        The full filepath to the URDF robot description
    port : str
        The device name for the serial port
-->
<launch>
    <arg name="model" default="$(find curio_description)/urdf/curio.urdf.xacro" />
    <arg name="port" default="/dev/ttyUSB0" />
    <arg name="classifier_filename" default="$(find lx16a)/data/lx16a_tree_classifier.joblib" />
    <arg name="classifier_window" default="10" />
    <arg name="regressor_filename" default="$(find lx16a)/data/lx16a_tree_regressor.joblib" />

    <!-- LX16A encoder filter server -->
    <node pkg="lx16a" type="lx16a_encoder_filter_server" name="lx16a_encoder_filter_server"
        respawn="true" output="screen">
    </node>

    <!-- Controller for the mobile base  -->
    <node pkg="curio_base" type="rover_base_controller" name="curio_base_controller"
        respawn="true" output="screen">
        <param name="classifier_filename" value="$(arg classifier_filename)" />
        <param name="classifier_window" value="$(arg classifier_window)" />
        <param name="regressor_filename" value="$(arg regressor_filename)" />
        <rosparam command="load" file="$(find curio_base)/config/base_controller.yaml" />
        <rosparam subst_value="true">
            port: $(arg port)
        </rosparam>
    </node>

    <!-- Controllers and localisation -->
    <include file="$(find curio_control)/launch/control.launch" />

</launch>
